vllm/config.py

```diff
@@ -3935,6 +3947,9 @@ def with_hf_config(
     def __post_init__(self):
         """Verify configs are valid & consistent with each other.
         """
+        # remove in v0.10.x
+        self.decoding_config = self.structured_output_config
```

Comment:
Could you make this a deprecated property so users get warned (similar to `guided_decoding_backend` in what's now `StructuredOutputConfig`)